testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java

```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
      * block until the rest show up. If there are enough, they will
      * be returned immediately. The returned items will be cleared from the
      * accumulator.
-     * @param desiredQuantity Number of items being awaited
      * @return A list of desiredQuantity many items
      * @throws RuntimeException On failure to attain the requested number of events
      *                          within a reasonable waiting period
      */
     @NonNull
-    public List<HubEvent<?>> take(int desiredQuantity) {
-        // If we haven't yet received the desired quantity of events on the subscription,
-        // setup a latch to await the desired quantity, less the number of existing events.
-        // For example: I desire 5, I already have 3, I wait for 2 more.
-        if (events.size() < desiredQuantity) {
-            latch = new CountDownLatch(desiredQuantity - events.size());
-            Latch.await(latch);
-            latch = null;
-        }
+    public List<HubEvent<?>> takeAll() {
+        Latch.await(latch);
 
         // If we already had the right number of events,
         // or if our latch counted down as a result of receiving the remaining number of events,
         // return the requested number of events.
-        List<HubEvent<?>> returning = new ArrayList<>(events.subList(0, desiredQuantity));
+        CopyOnWriteArrayList<HubEvent<?>> returning = new CopyOnWriteArrayList<>(events);
```

Comment:
Whoa any reason to do this? Will a user ever need to write anything on the returned list? Can't you just return an immutable list of hub events?

Reply:
Good point!